### DISCUSSION TITLE
Safe to use seter function inside the dependency array of useEffect

### DISCUSSION THREAD

**Discussion by mbretter** [pmndrs/zustand#308] [Q&A]
Hi,

is it safe to use a seter function as a dependency when using an useEffect hook? 
Or is it generating a rendering loop, because of the changing function handles on every render?

Something like (from the examples):
``` 
function Controls() {
  const increasePopulation = useStore(state => state.increasePopulation)
  const bears = useStore(state => state.bears)

  useEffect(() => {
    increasePopulation();
  },[increasePopulation]);

  return <span>{bears}</span>
}
```
thx,
cheers

**Comment by dai-shi** [comment]
Yes, it's safe. It's created outside render and thus stable.
